godot-core/src/builtin/script.rs

```diff
@@ -0,0 +1,480 @@
+        let name = StringName::from_arg_ptr(p_name as GDExtensionTypePtr, PtrcallType::Standard);
+
+        let result = (*instance)
+            .inner
+            .borrow()
+            .get(name.clone())
+            .map(|variant| {
+                variant.move_return_ptr(
+                    r_ret as GDExtensionTypePtr,
+                    godot_ffi::PtrcallType::Standard,
+                )
+            })
+            .is_some();
+
+        result as u8
+    }
+
+    pub(super) unsafe extern "C" fn get_property_list_func<T: ScriptInstance>(
+        p_instance: GDExtensionScriptInstanceDataPtr,
+        r_count: *mut u32,
+    ) -> *const GDExtensionPropertyInfo {
+        let instance = p_instance as *const ScriptInstanceData<T>;
+
+        let property_list: Vec<_> = (*instance)
+            .inner
+            .borrow()
+            .property_list()
+            .iter()
+            .map(|prop| prop.property_sys())
+            .collect();
+
+        *r_count = property_list.len() as u32;
+
+        let ptr = Box::into_raw(property_list.into_boxed_slice());
+
+        (*ptr).as_ptr()
+    }
+
+    pub(super) unsafe extern "C" fn get_method_list_func<T: ScriptInstance>(
+        p_instance: GDExtensionScriptInstanceDataPtr,
+        r_count: *mut u32,
+    ) -> *const GDExtensionMethodInfo {
+        let instance = p_instance as *const ScriptInstanceData<T>;
+
+        let method_list: Vec<_> = (*instance)
+            .inner
+            .borrow()
+            .method_list()
+            .iter()
+            .map(|method| method.method_sys())
+            .collect();
+
+        *r_count = method_list.len() as u32;
+
+        let ptr = Box::into_raw(method_list.into_boxed_slice());
+
+        (*ptr).as_ptr()
+    }
+
+    pub(super) unsafe extern "C" fn free_property_list_func<T: ScriptInstance>(
+        p_instance: GDExtensionScriptInstanceDataPtr,
+        p_prop_info: *const GDExtensionPropertyInfo,
+    ) {
+        let instance = p_instance as *const ScriptInstanceData<T>;
+        let length = (*instance).inner.borrow().property_list().len();
+
+        let vec = Vec::from_raw_parts(p_prop_info as *mut GDExtensionPropertyInfo, length, length);
+
+        drop(vec);
+    }
+
+    pub(super) unsafe extern "C" fn call_func<T: ScriptInstance>(
+        p_self: GDExtensionScriptInstanceDataPtr,
+        p_method: GDExtensionConstStringNamePtr,
+        p_args: *const GDExtensionConstVariantPtr,
+        p_argument_count: GDExtensionInt,
+        r_return: GDExtensionVariantPtr,
+        r_error: *mut GDExtensionCallError,
+    ) {
+        let instance = p_self as *mut ScriptInstanceData<T>;
+        let method =
+            StringName::from_arg_ptr(p_method as GDExtensionTypePtr, PtrcallType::Standard);
+
+        let args: Vec<_> = slice::from_raw_parts(p_args, p_argument_count as usize)
+            .iter()
+            .map(|ptr| &*Variant::ptr_from_sys(force_mut_ptr(*ptr)))
```

Comment:
Could you use `Variant::unbounded_refs_from_sys()` here?
If yes, the `#[cfg(since_api = "4.2")]` can be removed.

If not, `Variant::ptr_from_sys_mut()` followed by `to_const_ptr` would be preferred over `force_mut_ptr`.

Reply:
I will look into it. I implemented this a while ago and the internals probably changed since then.